hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsNetworkStatistics.java

```diff
@@ -231,7 +237,17 @@ public void testAbfsHttpResponseStatistics() throws IOException {
       // 1 read request = 1 connection and 1 get response
       expectedConnectionsMade++;
       expectedGetResponses++;
-      expectedBytesReceived += bytesWrittenToFile;
+      if (!getConfiguration().getHeadOptimizationForInputStream()) {
+        expectedBytesReceived += bytesWrittenToFile;
+      } else {
+        /*
+         * With head optimization enabled, the abfsInputStream is not aware
+         * of the contentLength and hence, it would only read data for which the range
+         * is provided. With the first remote call done, the inputStream will get
+         * aware of the contentLength and would be able to use it for further reads.
+         */
+        expectedBytesReceived += 1;
```

Comment:
With head optimization enabled and read fullFile optimization enabled, the abfsInputStream is not aware of the contentLength and hence, it would only read data for which the range is provided. With the first remote call done, the inputStream will get aware of the contentLength and would be able to use it for further reads.

Reply:
That should mean +1 for operation, why would the bytes received be 1. Bytes received should be equal to the range it read right

Reply:
At this point, the inputStream is at position 0 and the read request from application is 1 Byte. If the read full-file optimization is enabled, the inputStream would attempt to read the first readBuffer block from the file, which would read the whole file as the fileContentLength is smaller than the readBuffer size.